### ISSUE TITLE
Remove trailing slash from page URLs

### ISSUE THREAD

**Issue by xyeLz** [retypeapp/retype#393]
It is my understanding that for SEO, pages should _not_ have trailing slashes. Would it be possible to remove trailing slashes on URLs where applicable?

**Comment by geoffreymcgill** [comment]
Please add the following to your projects `retype.yml` file:

```yml
generator:
  trailingSlash: false
```

Hope this helps.

**Comment by xyeLz** [comment]
Thanks! Will the page load without the slash and then redirect to the page with the slash?

**Comment by xyeLz** [comment]
This seems to cause weird formatting issues with my site. I suspected a possible caching issue but even when accessing the site directly using https://d073989f.retypetest.pages.dev it still seems to cause the issue. For example, the logo in the top left appears initially but then disappears. Occasionally/randomly, my site loses all CSS formatting and just becomes a basic HTML page.

**Comment by geoffreymcgill** [comment]
I was able to reproduce with your live site. There is something strange happening with how the paths to the static resources are being emitted. I'll need to run some tests and see if I can reproduce locally.

**Comment by geoffreymcgill** [comment]
Sorry, but I've lost track of where your public repo for this content is located. Can you share a link to the repo again?

**Comment by geoffreymcgill** [comment]
Yes, there appears to be an issue with how the paths are being resolved for the static resources when `trailingSlash: false` is set in the project `retype.yml` file.

For now, please remove `trailingSlash: false` or set to `true`. 

```
# generator:
#   trailingSlash: false
```

We are investigating.

**Comment by xyeLz** [comment]
> Sorry, but I've lost track of where your public repo for this content is located. Can you share a link to the repo again?

Sure! It’s https://github.com/alukosdev/retypetest

**Comment by KrystianHutyra** [comment]
Is there any new information on this topic?

When I run retype locally with "trailingSlash: false" the page does not have the "/" character at the end, but when I run the page on Github Pages the "/" still appears. It seems that Github ignores this setting

**Comment by geoffreymcgill** [comment]
> When I run retype locally with `trailingSlash: false` the page does not have the `/` character at the end, but when I run the page on Github Pages the `/` still appears. It seems that Github ignores this setting

It looks like GitHub has enabled the following setting on their web servers, which automatically redirects `/sample​` to `/sample/`​. 

https://httpd.apache.org/docs/2.2/mod/mod_dir.html

If you request the following page without a trailing slash, the server sends a redirect​ response. The browser then re-requests the page with the trailing slash included. 

https://retypeapp.github.io/testing/brands/alpha

<img width="399" alt="Image" src="https://github.com/user-attachments/assets/fb6d200f-315b-4016-a6bc-8041c47c8e33" />

Others have inquired about the same issue, see:

https://stackoverflow.com/questions/33270605/github-pages-trailing-slashes

Best I can tell, there is no way to change or work-around this GitHub server setting. It is something GitHub forces when using their hosting service, although I'm not sure why.

If using GitHub Pages for hosting, setting `trailingSlash: false`​ will cause extra browser requests to be sent. The performance of the website will be better on GitHub Pages if you remove that setting from your `retype.yml​`.
